utils/users.py

```diff
@@ -108,5 +148,29 @@ async def get_current_user(token: str = Depends(oauth2_scheme)):
             status_code=status.HTTP_404_NOT_FOUND,
             detail="Could not find user",
         )
+    if not user.is_active:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="User is inactive",
+            headers={"WWW-Authenticate": "Bearer"},
+        )
 
     return users_schema.SystemUser(**user)
+
+
+async def get_admin_user(token: str = Depends(oauth2_scheme)):
+    """ Get auth user for validation """
+
+    users = await users_crud.filter_users(users_schema.UserFilter(**{}))
```

Comment:
why do you need to get all users here?

Reply:
To check if that user is first. Then he will be admin. After that all user by default will be without admin rights and admin will give change it manually by update functionality.